### PR TITLE
First Draft of Simplex Integration

### PR DESCRIPTION
**Pull request by wbobeirne**
This is a first draft of the simplex integration purchase page. This PR should be used for both technical and design feedback.

## Technicals

A requirement from the simplex team is that the page be handed off to them as simple HTML and CSS. This made using a framework like React or a JS-based bundler like Webpack unappealing. I stuck with a dead simple grunt script that uses EJS and SASS to generate the HTML and CSS respectively. It's not a particularly sophisticated build process, but it works.

I also tried to keep this page as lean and quick as possible, so it's entirely our own styling with no frameworks. Bootstrap, Vue, and jQuery JS libraries are included as those are required based on Simplex's scripts. What little additional Javascript was needed is just plopped into <script> tags in the relevant components.

When you build the site, you'll see that a lot of the JS is missing or broken. This is because Simplex has not provided us with those scripts yet, instead opting to have us send the HTML and CSS to them, and allowing them to wire it up.

This is by no means a beautifully organized project, my plan was to keep it simple, and make it easy for anyone to alter as needed.

## Design

Simplex provided us a few examples from [buy.bitcoin.com](https://buy.bitcoin.com), [buybitcoin.shapeshift.com](https://buybitcoin.shapeshift.com), and [buy.bitwala.com](https://buy.bitwala.com). I tried to keep it as simple as those, while still maintaining our styles and branding from the beta site. I also incorporated bitcoin.com's FAQ. At the bottom is the same footer on live and beta, but somewhat stripped down to the basics. All of the copy is subject to change, pending review from Jordan and Taylor.

Here's a screenshot of the full page, but note that the scripts aren't working, and they haven't switched it over to ETH from BTC, so some things aren't looking quite right:

image

And here it is on a mobile screen:

screen shot 2018-03-17 at 5 18 38 pm

## Running it Yourself

You can find all of the necessary information for running this in the readme. Any and all feedback would be appreciated.

### CODE REVIEW

File: src/scss/components/app.scss [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    opacity: 0;
+  }
+  100% {
+    transform: translateY(5px);
+    opacity: 1;
+  }
+}
+
+// All inputs look the same
+input {
```
Comment (wbobeirne) [inline src/scss/components/app.scss:18]: This should be an `.Input` class or something in a separate file.

File: src/ejs/includes/app.ejs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+<%# control JS end %>
+
+<%# custom JS begin %>
+<script>
+  // Pull address out of query, stick in input
+  (function() {
+    const queryMatches = window.location.href.match(/[?&]address=([0-9a-zA-Z]+)/i);
+    if (queryMatches && queryMatches[1]) {
+      const addrInput = document.querySelector('.App-address-input');
+      addrInput.value = queryMatches[1];
```
Comment (dternyak) [inline src/ejs/includes/app.ejs:138]: any possibility this could lead to xss? (I'll circle back and answer my own question)

Comment (409H) [reply src/ejs/includes/app.ejs:138]: Nope - the possible values are alphanumeric as per the `match()` on line 135

Comment (wbobeirne) [reply src/ejs/includes/app.ejs:138]: @409H has it, this only picks up `0-9a-zA-Z`. I was very concerned about XSS after all of our recent struggles, haha.

**Comment by jspence425** [comment]
Is the fee $10 minimum, or is the actual transaction itself $10 minimum? Also, it should be 3% to MyCrypto.

What's the easiest way to see the rest of the FAQ?

**Comment by 409H** [comment]
> What's the easiest way to see the rest of the FAQ?

@jspence425 Looks like all the FAQs are in this file: https://github.com/MyCryptoHQ/buy.mycrypto.com/blob/draft/src/data.js

File: src/data.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    `
+  }, {
+    q: 'Who provides this service?',
+    a: `
+      The purchasing process is handled by our partners at Simplex.com. This
+      page is hosted by MyCrypto.
+    `
+  }, {
+    q: 'Are there any fees?',
+    a: 'There is a Simplex fee of 5% ($10 minimum) and a 2% tip to MyCrypto.'
```
Comment (wbobeirne) [inline src/data.js]: Up MyCrypto tip to 3%.

Comment (wbobeirne) [reply src/data.js]: Potentially make it clearer that the fee is 5% OR $10, whichever is higher.

Comment (jspence425) [reply src/data.js]: Ok cool. I'll submit a PR (I think that's the right term) for any changes this evening ideally.

**Comment by 409H** [comment]
This may just be placeholders, but it says you can buy Ether, but you need to provide Bitcoin details - should this be changed to ETH and Ether address?

image

**Comment by wbobeirne** [comment]
@409H 

> Here's a screenshot of the full page, but note that the scripts aren't working, and they haven't switched it over to ETH from BTC, so some things aren't looking quite right:

The Simplex team has said that they'll do that switchover for us. I wish they'd just tell me how, but ciest la vie.

**Comment by wbobeirne** [comment]
Merging this PR as-is to keep the ball rolling with Simplex. Feel free to drop additional comments though, or open issues against this repo.

**Pull request merged**